src/gui/layout/qgslayoutlegendwidget.h

```diff
@@ -33,6 +33,36 @@ class QgsLayoutLegendMapFilteringWidget;
 
 ///@cond PRIVATE
 
+#ifndef SIP_RUN
+/**
+ * \ingroup core
+ * \brief A layout item subclass for map legend filtering.
+ * \note Not available in Python bindings
+ * \since QGIS 4.0
+ */
+class GUI_EXPORT QgsLegendLayerTreeProxyModel : public QgsLayerTreeProxyModel
+{
+    Q_OBJECT
+  public:
+    /**
+     * Constructor for QgsLegendLayerTreeProxyModel, with the specified \a parent object.
+     */
+    QgsLegendLayerTreeProxyModel( QgsLayoutItemLegend *legend, QObject *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Sets whether the legend is showing the default legend for a project (as opposed
+     * to a customised legend).
+     */
+    void setIsDefaultLegend( bool isDefault );
```

Comment:
No getter needed?

Reply:
This class is just a private class -- so we an add a getter when/if needed